pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -165,6 +165,8 @@ public class MutableSegmentImpl implements MutableSegment {
   private final PartitionUpsertMetadataManager _partitionUpsertMetadataManager;
   private final List<String> _upsertComparisonColumns;
   private final String _deleteRecordColumn;
+  private final String _upsertOutOfOrderRecordColumn;
```

Comment:
We can add a validation in `TableConfigUtils` to verify only one of `_upsertOutOfOrderRecordColumn` and `_upsertDropOutOfOrderRecord` should be set